specs/ScreenCaptureRequested.md

```diff
@@ -0,0 +1,357 @@
+# Background
+
+The HTML DOM's Screen Capture API `navigator.mediaDevices.getDisplayMedia` allows developers to 
+get a video stream of a user's tabs, windows, or desktop. This API is available in WebView2, 
+but the current default UI has some problems that we need to fix to make sure that hybrid apps 
+using WebView2 have a more seamless web/native experience. This includes removing the tab column 
+in the UI, replacing default strings and icons that do not match in WV2, and potentially having 
+the ability to customize the UI itself. 
+
+These apps also expect that the screen capture dialog has an event before the UI is shown to give 
+the host app an opportunity to block or allow UI from showing at all.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+
+We propose introducing the `ScreenCaptureRequested` event. This event will be raised whenever 
+the WebView2 and/or iframe corresponding to the CoreWebView2Frame or any of its descendant iframes 
+requests permission to use the Screen Capture API before the UI is shown. 
+
+For convenience of the end developer, by default we plan to raise
+`ScreenCaptureRequested` on both `CoreWebView2Frame` and `CoreWebView2`. The
+`CoreWebView2Frame` event handlers will be invoked first,
+before the `CoreWebView2` event handlers. If `Handled` is set true as part of
+the `CoreWebView2Frame` event handlers, then the `ScreenCaptureRequested` event
+will not be raised on the `CoreWebView2`, and its event handlers will not be
+invoked.

```

Comment:
Are there really 4 cases that need to be supported (the combinations of `Handled` and `Cancel`)? I think if you set `Cancel` you're always going to also set `Handled`. And if you don't set `Cancel` I think you're not going to set `Handled`.

I.e., I don't think `Handled` is necessary; if `Cancel` is set, then stop raising events.

Reply:
On the other hand, the double boolean is clearer that one of them controls propagation and the other controls the system behavior. It also aligns with other uses of Handled and Cancel.

You might set Handled=true and Cancel=false to mean "Trust me, it's okay. Don't need to ask others for permission to capture. I give you permission to capture."

And you might set Handled=false and Cancel=true to mean "I think you should cancel, but call the other handlers because one of them might have a stronger opinion."

Reply:
Leave as is: match existing xaml / dom patterns of bubbling